src/styles/main.scss

```diff
@@ -1,3 +1,68 @@
+$count-tag:31;
+
+$start-day:
+"mon" 0, "tue" 1, "wed" 2, "thu" 3, "fri" 4, "sat" 5, "sun" 6;
+
+@mixin square {
+  width: 100px;
+  height: 100px;
+}
+
 body {
   margin: 0;
 }
+
+.body {
+  display: flex;
+  height: 100vh;
+  align-items: center;
+  justify-content: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  padding: 10px;
+  width: 707px;
+
+  @each $start-day, $count in $start-day {
+    &--start-day--#{$start-day} :first-child {
+      margin-left: $count * 100px + $count;
+    }
+  }
+
+  @for $tag from 28 through $count-tag {
+    &--month-length--#{$tag} :nth-child(n + #{$tag + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    box-sizing: border-box;
+    transition: all 0.5s ease 0s;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+
+  @for $index from 1 through $count-tag {
+    &__day--#{$index} {
+      @include square;
+      border: 1px solid black;
+      background-color: #eee;
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+
+      &::before {
+        content: "#{$index}";
+      }
```

Comment:
leave only `before` in this loop. others are just common styles and no need to repeat them
```
&::before {
        content: "#{$index}";
      }
```